core/domain/collection_domain.py

```diff
@@ -1145,6 +1152,35 @@ def delete_node(self, exploration_id):
                 exploration_id)
         del self.nodes[node_index]
 
+    def move_node(self, exploration_id, new_position):
+        """Changes the position of the node corresponding to the given
+        exploration in the nodes list of the collection.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+            new_postion: int. The new position of the node in the nodes list.
+
+        Raises:
+            ValueError: The exploration is not part of the collection.
+            ValueError: The exploration cannot be moved to the new_position or
+            it's already there'.
+        """
+        old_position = self._find_node(exploration_id)
```

Comment:
Just to double-check, do we have a validation check already (on-save) that ensures that a collection can't contain multiple nodes with the same exploration ID?